src/routes/userRouter.ts

```diff
@@ -30,19 +36,46 @@ const signInSchema = z.object({
 userRouter.post("/signUp", zValidator("json", signUpSchema), async (c) => {
     const { fullName, email, password } = await c.req.valid("json");
     const hashedPassword = await Bun.password.hash(password);//hashing password
+    const hashedToken = await bcryptjs.hash(email, 10);
+
+    console.log("Hashed token:", hashedToken);
+    
 
     try {
         await db.insert(usersTable).values({
             fullName,
             email,
             passwordHash: hashedPassword,
+            verificationToken: hashedToken,
+            verificationTokenExpiry: new Date(Date.now() + 3600000) // 1 hour from now
         });
 
-        return c.json({ message: "User signed up successfully" });
-    } catch (e) {
+        // Looking to send emails in production? Check out our Email API/SMTP product!
+var transport = nodemailer.createTransport({
```

Comment:
Extract out the email sending part into a function which can live in `src/lib/email` folder or `src/lib/email.ts` file